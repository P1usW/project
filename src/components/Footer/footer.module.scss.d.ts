declare namespace FooterModuleScssNamespace {
  export interface IFooterModuleScss {
    footer: string;
    footer_icon: string;
    footer_link: string;
    footer_span: string;
    footer_text: string;
    'icon-resize': string;
  }
}

declare const FooterModuleScssModule: FooterModuleScssNamespace.IFooterModuleScss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: FooterModuleScssNamespace.IFooterModuleScss;
};

export = FooterModuleScssModule;
