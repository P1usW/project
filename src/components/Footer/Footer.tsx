import { FC } from 'react';
import style from './footer.module.scss';
import { Icon } from '@iconify/react';

const Footer: FC = () => {
  return (
    <footer className={style.footer}>
      <div
        className={style.footer_text}
      >
        <p>
          <span className={style.footer_span}>Iv</span>an <span className={style.footer_span}>Sh</span>avrin.
        </p>
        <p>
          Frontend - developer.
        </p>
      </div>
      <div 
        className={style.footer_text}
      >
        <p>My contacts are on top</p>
        <a
          onClick={() => window.scrollTo(0, 0)}
          className={style.footer_link}
        >
          back to top 
        </a>
      </div>
      <div
        className={style.footer_text}
      >
        <p>
          The site was created without using UI libraries, if you find a flaw, you can leave a message on <i>github</i> <a className={style.footer_icon} target='_blank' rel='noreferrer' href='#'><Icon icon="bx:edit"/></a>
        </p>
        <p>
          Â© 2023 IvSh. All Rights Reserved.
        </p>
      </div>
    </footer>
  );
};

export default Footer;
