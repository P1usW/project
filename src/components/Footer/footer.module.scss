.footer {
  box-sizing: border-box;
  display: grid;
  gap: 15px;
  grid-template-areas: 
    "name other"
    "info info";
  grid-template-columns: repeat(2, minmax(200px, 1fr));

  padding: 25px;
  width: 100%;
  height: 10rem;

  
  & :first-child {
    grid-area: name;
  }

  & :nth-child(2) {
    grid-area: other;
  }

  & :nth-child(3) {
    grid-area: info;
  }


  &_span {
    font-weight: 800;
  }

  &_text {
    font-size: small;
    text-align: center;
  }

  &_icon {
    cursor: pointer;
    color: currentColor;
    display: inline-block;
    vertical-align: middle;
  }

  &_link {
    color: var(--primary-color);
    text-decoration: underline;
    cursor: pointer;
  }
}
