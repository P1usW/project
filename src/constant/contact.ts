export default {
  email: 'mailto:ivshavrin@gmail.com',
  vk: 'https://vk.com/ivan_4cb',
  gitHub: 'https://github.com/P1usW',
  telegram: 'https://t.me/Ivan_4CB',
  linkedin: 'https://www.linkedin.com/in/p1usw/',
}