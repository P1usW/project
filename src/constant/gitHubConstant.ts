const baseURL = 'https://api.github.com';
const axiosConfig = {
  method: 'get',
  baseURL,
  timeout: 3000,
  headers: {
    'X-GitHub-Api-Version': '2022-11-28'
  },
}


export {
  baseURL,
  axiosConfig
}