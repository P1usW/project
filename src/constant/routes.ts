import Home from "@src/components/Home/Home";
import { defer } from "react-router";
import { getUserInfo } from "@src/api/GitHubApi";
import { lazy } from "react";

const Projects = lazy(() => import(/* webpackChunkName: "Projects" */ '@src/components/Projects/Projects'));
const NotFound = lazy(() => import(/* webpackChunkName: "NotFound" */ '@src/components/NotFound/NotFound'));
const About = lazy(() => import(/* webpackChunkName: "About" */ '@src/components/About/About'));

const publicRoutes = [
  {
    path: '/',
    Component: Home
  },
  {
    path: 'about',
    Component: About 
  },
  {
    path: 'projects',
    Component: Projects,
    loader: () => {
      const user = getUserInfo('Pl1usW');
      return defer({user})
    }
  },
  {
    path: '*',
    Component: NotFound
  }
];

export {
  publicRoutes,
}