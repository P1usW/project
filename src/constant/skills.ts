export const skills = {
  language: [
    'Python',
    'JavaScript',
    'TypeScript'
  ],
  frontend: [
    'HTML',
    'CSS / SASS / SCSS',
    'React / Vue',
    'Redux',
    'Webpack'
  ],
  backend: [
    'NodeJS',
    'Django',
    'PostgreSQL',
    'Docker',
    'Redis',
  ],
  other: [
    'Git',
    'Jest / TestingLibrary',
    'Figma',
    'Firebase'
  ]
}
