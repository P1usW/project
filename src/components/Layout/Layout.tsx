import { Outlet } from "react-router";
import NavBar from "../NavBar/NavBar";
import Footer from "../Footer/Footer";
import { FC, Suspense } from "react";
import { ScrollRestoration } from "react-router-dom";

const Layout: FC = () => {
  return (
    <>
      <NavBar/>
      <main>
        <Suspense fallback={<div>loading</div>}>
          <Outlet/>
        </Suspense>
      </main>
      <Footer/>
      <ScrollRestoration/>
    </>
  );
};

export default Layout;