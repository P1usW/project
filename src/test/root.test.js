import { fireEvent, render, screen } from '@testing-library/react';
import Footer from '@src/components/Footer/Footer';
import userEvent from '@testing-library/user-event';
import { 
  createMemoryRouter,
  RouterProvider,
} from 'react-router-dom';
import { publicRoutes } from '@src/constant/routes';
import Layout from '@src/components/Layout/Layout';
import '@testing-library/jest-dom';
import './testUtils/mokScrollTo'
import axios from 'axios';

jest.mock('../api/gitHubApi/getReposInfo', () => {
  return {
    __esModule: true,
    getReposInfo: jest.fn(() => [
      {
        key: 'repos.name',
        name: 'repos.name',
        html_url: 'repos.html_url'
      }
    ])
  }
})


describe.skip('Test render', () => {
  test('Footer snapshot', () => {
    render(<Footer/>);

    const footer = screen.getByRole('contentinfo')
    expect(footer).toMatchSnapshot();

    screen.debug();
  })
})

describe.skip('Test theme toggle', () => {
  let theme;

  beforeAll(() => {
    window.localStorage.getItem('theme');
  })

  test('Click theme button', async () => {
    // render(NavBar)

    const btn = screen.getByRole('button');
    // fireEvent.click(btn);
    await userEvent.click(btn);
  })
})

describe('Test navigation', () => {
  const router = createMemoryRouter([
    {
      Component: Layout,
      children: publicRoutes,
    }
  ]);

  render(
    <RouterProvider router={router}/>
  )
  
  afterEach(() => {
    jest.resetAllMocks()
  })
  afterAll(() => {
    jest.clearAllMocks()
  })

  test('has main page', () => {
    screen.getAllByRole('navigation');
    screen.debug();
  })
  
})