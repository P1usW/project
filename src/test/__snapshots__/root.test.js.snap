// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Footer snapshot 1`] = `
<footer
  class="footer"
>
  <div
    class="footer_text"
  >
    <p>
      <span
        class="footer_span"
      >
        Iv
      </span>
      an 
      <span
        class="footer_span"
      >
        Sh
      </span>
      avrin.
    </p>
    <p>
      Frontend - developer.
    </p>
  </div>
  <div
    class="footer_text"
  >
    <p>
      My contacts are on top
    </p>
    <a
      class="footer_link"
      href="#top"
    >
      back to top
    </a>
  </div>
  <div
    class="footer_text"
  >
    <p>
      The site was created without using UI libraries, if you find a flaw, you can leave a message on 
      <i>
        github
      </i>
       
      <a
        class="footer_icon"
        href="#"
        rel="noreferrer"
        target="_blank"
      >
        <span />
      </a>
    </p>
    <p>
      Â© 2023 IvSh. All Rights Reserved.
    </p>
  </div>
</footer>
`;
